### DISCUSSION TITLE
Setting correspondent, title and tags from filename

### DISCUSSION THREAD

**Discussion by GreenTeaBalls** [jonaswinkler/paperless-ng#1614] [General]
Hi @jonaswinkler,

first off - thank you very much for picking up the work on Paperless, I was delighted to see that someone continues to develop this piece of software!

I have tried Paperless in the past and after a brief trial period dropped it again, mainly due to the fact that getting the stored documents exposed through a standard filesystem to the non-IT-savvy people in my household proved complex and troublesome.

I currently keep ~1600 ocr'd PDF documents on a fileshare. They are named using the following format {2022-02-08 - Correspondent - Title,tag,tag,tag.pdf}. You may be able to see that I took inspiration from Paperless' original naming format.

Is there any way to get these documents into Paperless without having to set date (which often doesn't correspond with creation date of the file as I have been scanning in older documents), correspondent and title (and maybe tags) from the filename?

The idea of having to manually add date, correspondent, title and tags to all those 1600 documents in Paperless to have the files written to disk again with a human-readable filename is prohibitive and keeps me from using Paperless (which I'd very much like to now that you have addressed many other issues I had with it before).

Any ideas most welcome. Thank you!

**Comment by rsmuc** [comment]
For the date you could use the `PAPERLESS_FILENAME_DATE_ORDER=<format>` parameter.  Then paperless will use first the date from the filename and then look into the content of the document. (Paperless will not use the creation date of the PDF as far as I know)
More details here: https://paperless-ng.readthedocs.io/en/latest/configuration.html

Setting the Correspondent etc. automatically could be done via the API with a post-consumption script. But that's a little more tricky.
An example could be found here (in German, but maybe google translate will help you): https://paperless.codeberg.page/paperless/2021-06/paperless-libreoffice/#automatische-klassifizierung 

Extended example: https://paperless.codeberg.page/paperless/2021-12-2/paperless-date-validation/#die-l%C3%B6sung

And here are some more details about the REST API: https://paperless-ng.readthedocs.io/en/latest/api.html

**Reply by GreenTeaBalls** [reply]
Thank you very much @rsmuc, your pointers are most appreciated. I think with the ideas you have given me I should be able to rustle up some bash or maybe even python script to help me get this working.

Your hint for the `PAPERLESS_FILENAME_DATE_ORDER` parameter already solved the first issue with getting the date from the filename.
